command/cp.go

```diff
@@ -113,6 +119,14 @@ var copyCommandFlags = []cli.Flag{
 		Value:   defaultPartSize,
 		Usage:   "size of each part transferred between host and remote server, in MiB",
 	},
+	&cli.StringFlag{
+		Name:  "source-region",
+		Usage: "connect to a specific region of the remote object storage service",
+	},
+	&cli.StringFlag{
```

Comment:
Why two flags are needed for cross-region support? We either create source or destination client. `awscli` only has `          [--source-region <value>]`  flag to support this functionality. Maybe you may want to investigate how they achieve this.

Reply:
sorry for the confusion, similar flag options are [here](https://github.com/aws/aws-cli/issues/2274#issuecomment-258010608). Normally, by default s5cmd takes region as us-east-1 by default, so if some other region is desired we can provide using --source-region flag. it can be useful for scenario where we have many operations and total number of unique regions (source+destination) are more than 2. e.g, (us-west-1 -> us-north-1) and (eu-east-1 -> eu-south-2)

Reply:
The reason why this part confused me is that we can already define source aws region with either environment variables or aws profile. That is why I thought it would be enough to add only one flag for the destination region of S3 object.

```
AWS_REGION=ap-northeast-2 s5cmd ls s3://asiatest
AWS_REGION=ap-northeast-2 s5cmd cp --region us-east-1 s3://asiatest/object s3://as
```

or
```
// .aws/config
[asia]
region=ap-northeast-2 
[east]
region=us-east-1
```

I guess this flag is only needed for batch mode since all operations will share same AWS profile and environment variables if we don't provide an optional flag for source region. Thanks for your clarification. Could you please add e2e test and update the documentation?